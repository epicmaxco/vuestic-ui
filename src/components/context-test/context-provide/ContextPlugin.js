import flow from 'lodash/flow'
import camelCase from 'lodash/camelCase'
import upperFirst from 'lodash/upperFirst'
import Vue from 'vue'

const pascalCase = flow(camelCase, upperFirst)

/**
 * Key for communication inject/provide
 */
export const ContextProviderKey = 'va-context-provider'

/**
 * Plugin provide global config to Vue component through prototype
 */
export const ContextPlugin = {
  install (Vue, options = {}) {
    Vue.prototype.$vaContextConfig = Vue.observable(options)
  },
}

/**
 * Mixin provide local configs to Vue component through injecting
 * All list of local configs contain in this._$configs
 */
export const ContextPluginMixin = {
  inject: {
    _$configs: {
      from: [ContextProviderKey],
      default: () => [],
    },
  },
}

/**
 * Calc value of property from component, local and global config
 *
 * @category String
 * @param context [object] this of the vue component.
 * @param prop [string=''] The string of property name.
 * @param defaultValue [any] The default property value.
 * This value takes when each local or global config and component do not contain property.
 * @returns {any} Returns property value.
 *
 */
export function getContextPropValue (context, prop, defaultValue) {
  // We have to pass context here as this method will be mainly used in prop default,
  // and methods are not accessible there.

  // Local prop takes priority even when empty.
  if (context.hasOwnProperty(prop)) {
    return context[prop]
  }

  const componentName = pascalCase(context.$options.name)

  if (!context._$configs) {
    throw new Error(`'getContextPropValue' working only together with 'ContextPluginMixin'. Please, use 'ContextPluginMixin' for ${componentName} component`)
  }

  const configs = context.$vaContextConfig ? [context.$vaContextConfig, ...context._$configs] : context._$configs
  const config = getLocalConfigWithComponentProp(configs, componentName, prop)

  return config ? config[componentName][prop] : defaultValue
}

// Allows to completely overwrite global context config.
export function overrideContextConfig (context, options) {
  // Clear object
  for (const key in context.$vaContextConfig) {
    Vue.delete(context.$vaContextConfig, key)
  }
  // Set values
  for (const key in options) {
    Vue.set(context.$vaContextConfig, key, options[key])
  }
}

/**
 * Attempt to find a prop value from config chain.
 *
 * @category String
 * @param configs [object] config chain.
 * @param componentName [string]
 * @param propName [string] property name (pascal-case)
 *
 * @returns {any} config object if found undefined means not found.
 */
function getLocalConfigWithComponentProp (configs, componentName, propName) {
  // Find prop value in config chain.
  return configs.reverse().find(config => {
    const componentConfig = config[componentName]
    return componentConfig && componentConfig.hasOwnProperty(propName)
  }) || undefined
}
