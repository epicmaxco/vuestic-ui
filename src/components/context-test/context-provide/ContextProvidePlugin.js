import flow from 'lodash/flow'
import camelCase from 'lodash/camelCase'
import upperFirst from 'lodash/upperFirst'
import cloneDeep from 'lodash/cloneDeep'
import assignWith from 'lodash/assignWith'

const pascalCase = flow(camelCase, upperFirst)
export const ContextProviderPluginKey = 'va-context-config'

class ContextProviderConfig {
  constructor (options) {
    this._config = options
  }
  getNewConfig (config) {
    const _combinedConfig = assignWith(config, cloneDeep(this._config), (objVal) => objVal) // NOTE: Merge external new config with current config. Need save the reference to config prop. Support reactivity

    return new ContextProviderConfig(_combinedConfig)
  }
  getComponentConfig (name, prop = null) {
    const _componentConfig = this._config[pascalCase(name)]

    return prop ? _componentConfig[prop] : _componentConfig
  }
}

export const ContextProvidePlugin = {
  install (Vue, options = {}) {
    Vue.prototype.$vaContextConfig = Vue.observable({
      [ContextProviderPluginKey]: new ContextProviderConfig(options),
    })
  },
}

export const ContextProvideMixin = {
  inject: {
    _$context: {
      from: [ContextProviderPluginKey],
      default: () => null,
    },
  },
  methods: {
    getProviderConfig (prop) {
      const currentConfigConstructor = this._$context || this.$vaContextConfig[ContextProviderPluginKey]

      return this[prop] || currentConfigConstructor.getComponentConfig(pascalCase(this.$options.name), prop)
    },
  },
}
