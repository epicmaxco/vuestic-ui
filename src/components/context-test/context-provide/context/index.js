import iconsDemoConfig from './va-icons-demo-config'
import iconsDocsConfig from './va-icons-docs-config'
import iconsFrameworkConfig from './va-icons-framework-config'

export const getContext = () => {
  return {
    VaIcon: {
      font: 'md',
      config: { ...iconsDemoConfig, ...iconsDocsConfig, ...iconsFrameworkConfig },
    },
  }
}
