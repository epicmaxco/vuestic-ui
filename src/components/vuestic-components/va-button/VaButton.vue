<template>
  <component
    :is="computedTag"
    class="va-button"
    :class="computedClass"
    :style="computedStyle"
    :disabled="disabled"
    :type="type"
    :href="href"
    :target="target"
    :to="to"
    :replace="replace"
    :append="append"
    :active-class="activeClass"
    :exact="exact"
    :exact-active-class="exactActiveClass"
    v-on="inputListeners"
    @mouseenter="updateHoverState(true)"
    @mouseleave="updateHoverState(false)"
    @focus="updateFocusState(true)"
    @blur="updateFocusState(false)"
    tabindex="0"
  >
    <div class="va-button__content">
      <va-icon
        v-if="icon"
        fixed-width
        class="va-button__content__icon va-button__content__icon-left"
        :name="icon"
      />
      <div
        v-if="hasTitleData"
        class="va-button__content__title">
        <slot/>
      </div>
      <va-icon
        v-if="iconRight"
        fixed-width
        class="va-button__content__icon va-button__content__icon-right"
        :name="iconRight"
      />
    </div>
  </component>
</template>

<script>
import VaIcon from '../va-icon/VaIcon'
import {
  getGradientBackground,
  getFocusColor,
  getHoverColor,
  getBoxShadowColor,
} from '../../../services/color-functions'
import { ColorThemeMixin } from '../../../services/ColorThemePlugin'

export default {
  name: 'va-button',
  components: { VaIcon },
  mixins: [ColorThemeMixin],
  inject: {
    va: {
      default: () => ({}),
    },
    contextConfig: {},
  },
  props: {
    tag: {
      type: String,
      default: 'button',
    },
    color: {
      type: String,
      default: 'primary',
    },
    outline: {
      type: Boolean,
    },
    flat: {
      type: Boolean,
    },
    small: {
      type: Boolean,
    },
    large: {
      type: Boolean,
    },
    icon: {
      type: String,
    },
    iconRight: {
      type: String,
    },
    type: {
      type: String,
    },
    disabled: {
      type: Boolean,
    },
    /* Link props */
    href: {
      type: String,
    },
    target: {
      type: String,
    },
    /* Router link props */
    to: {
      type: [String, Object],
    },
    replace: {
      type: Boolean,
    },
    append: {
      type: Boolean,
    },
    activeClass: {
      type: String,
    },
    exact: {
      type: Boolean,
    },
    exactActiveClass: {
      type: String,
    },
  },
  data () {
    return {
      hoverState: false,
      focusState: false,
    }
  },
  computed: {
    computedClass () {
      return {
        'va-button--default': !this.flat && !this.outline,
        'va-button--flat': this.flat,
        'va-button--outline': this.outline,
        'va-button--disabled': this.disabled,
        'va-button--hover': this.hoverState,
        'va-button--focus': this.focusState,
        'va-button--without-title': !this.hasTitleData,
        'va-button--with-left-icon': this.icon,
        'va-button--with-right-icon': this.iconRight,
        'va-button--large': this.large,
        'va-button--small': this.small,
        'va-button--normal': !this.large && !this.small,
      }
    },
    gradientStyle () {
      if (this.flat || this.outline) {
        return
      }
      if (this.va.color) {
        return
      }
      return getGradientBackground(this.colorComputed)
    },
    shadowStyle () {
      if (this.flat || this.outline || !this.contextConfig.shadow) {
        return
      }

      if (this.contextConfig.shadow === 'sm') {
        return '0 1px 1px 0 rgba(10, 13, 117, 0.25)'
      }

      if (this.contextConfig.shadow === 'lg') {
        if (this.va.color && this.$themes && this.$themes[this.va.color]) {
          return '0 0.125rem 0.19rem 0 ' + getBoxShadowColor(this.color ? this.colorComputed : this.$themes[this.va.color])
        }

        return '0 0.125rem 0.19rem 0 ' + getBoxShadowColor(this.colorComputed)
      }
      return {}
    },
    computedStyle () {
      const computedStyle = {
        color: '',
        borderColor: '',
        background: '',
        backgroundImage: '',
        boxShadow: '',
      }

      if (this.focusState) {
        if (this.outline || this.flat) {
          computedStyle.color = this.colorComputed
          computedStyle.borderColor = this.outline ? this.colorComputed : ''
          computedStyle.background = getFocusColor(this.colorComputed)
        } else {
          computedStyle.backgroundImage = this.gradientStyle
        }
      } else if (this.hoverState) {
        if (this.outline || this.flat) {
          computedStyle.color = this.colorComputed
          computedStyle.borderColor = this.outline ? this.colorComputed : ''
          computedStyle.background = getHoverColor(this.colorComputed)
        } else {
          computedStyle.backgroundImage = this.gradientStyle
        }
      } else {
        computedStyle.color = this.flat || this.outline ? this.colorComputed : '#ffffff'
        computedStyle.borderColor = this.outline ? this.colorComputed : ''
        computedStyle.backgroundImage = this.gradientStyle
        computedStyle.boxShadow = this.shadowStyle
      }

      if (
        !this.outline &&
        !this.flat &&
        (this.va.color || !this.contextConfig.gradient)
      ) {
        computedStyle.background = this.color ? this.colorComputed : this.$themes[this.va.color]
        computedStyle.backgroundImage = ''
      }

      return computedStyle
    },
    hasTitleData () {
      return this.$slots.default
    },
    computedTag () {
      if (this.tag === 'a' || this.href || this.target) {
        return 'a'
      }
      if (this.tag === 'router-link' || this.to || this.append || this.replace ||
        this.activeClass || this.exact || this.exactActiveClass) {
        return 'router-link'
      }
      return 'button'
    },
    inputListeners () {
      const vm = this
      return Object.assign({},
        this.$listeners,
        {
          click (event) {
            vm.$emit('click', event)
          },
        },
      )
    },
  },
  methods: {
    updateHoverState (isHover) {
      this.hoverState = isHover
    },
    updateFocusState (isHover) {
      this.focusState = isHover
    },
  },
}
</script>

<style lang='scss'>
@import "../../vuestic-sass/resources/resources";

.va-button {
  display: inline-block;
  margin: $btn-margin;
  background-image: none;
  box-shadow: none;
  outline: none !important;
  border: $btn-border;
  font-family: $font-family-sans-serif;
  text-decoration: none !important;
  text-transform: initial;
  font-weight: $font-weight-bold;
  cursor: pointer;
  transition: $btn-transition;
  background-color: $white;

  &__content {
    display: flex;

    &__title, &__icon {
      display: flex;
      justify-content: center;
      align-items: center;
      margin: auto;
    }
  }

  &--default {
    color: $white;

    &:hover {
      filter: brightness(115%);
    }

    &:focus, &:active {
      filter: brightness(85%);
    }

    i {
      color: $white;
    }
  }

  &--outline {
    background-color: transparent;
    border: solid $btn-border-outline;
    text-decoration: none;

    &.va-button--disabled {
      background: transparent;
      @include va-disabled;

      &.va-button--active {

        .va-button__content, i {
          color: $white !important;
        }
      }
    }
  }

  &--flat {
    background: transparent;
    border: $btn-border solid transparent;
    text-decoration: none;
  }

  &.va-button--disabled {
    @include va-disabled;
  }

  &--large {
    @include va-button($btn-padding-y-lg, $btn-padding-x-lg, $btn-font-size-lg, $btn-line-height-lg, $btn-border-radius-lg);
    letter-spacing: $btn-letter-spacing-lg;

    .va-button__content__icon {
      width: $btn-icon-width-lg;
    }

    &.va-button--with-left-icon {
      padding-left: $btn-with-icon-wrapper-padding-lg;

      &.va-button--without-title {
        padding-right: $btn-with-icon-wrapper-padding-lg;
      }

      .va-button__content__title {
        padding-left: $btn-with-icon-content-padding-lg;
      }
    }

    &.va-button--with-right-icon {
      padding-right: $btn-with-icon-wrapper-padding-lg;

      .va-button__content__title {
        padding-right: $btn-with-icon-content-padding-lg;
      }
    }

    &.va-button--outline {
      line-height: $btn-line-height-lg - 2 * $btn-border-outline;
    }
  }

  &--small {
    @include va-button($btn-padding-y-sm, $btn-padding-x-sm, $btn-font-size-sm, $btn-line-height-sm, $btn-border-radius-sm);
    letter-spacing: $btn-letter-spacing-sm;

    .va-button__content__icon {
      width: $btn-icon-width-sm;
    }

    &.va-button--with-left-icon {
      padding-left: $btn-with-icon-wrapper-padding-sm;

      &.va-button--without-title {
        padding-right: $btn-with-icon-wrapper-padding-sm;
      }

      .va-button__content__title {
        padding-left: $btn-with-icon-content-padding-sm;
      }
    }

    &.va-button--with-right-icon {
      padding-right: $btn-with-icon-wrapper-padding-sm;

      .va-button__content__title {
        padding-right: $btn-with-icon-content-padding-sm;
      }
    }

    &.va-button--outline {
      line-height: $btn-line-height-sm - 2 * $btn-border-outline;
    }
  }

  &--normal {
    @include va-button($btn-padding-y-nrm, $btn-padding-x-nrm, $btn-font-size-nrm, $btn-line-height-nrm, $btn-border-radius-nrm);
    letter-spacing: $btn-letter-spacing-nrm;

    .va-button__content__icon {
      width: $btn-icon-width-nrm;
    }

    &.va-button--with-left-icon {
      padding-left: $btn-with-icon-wrapper-padding-nrm;

      &.va-button--without-title {
        padding-right: $btn-with-icon-wrapper-padding-nrm;
      }

      .va-button__content__title {
        padding-left: $btn-with-icon-content-padding-nrm;
      }
    }

    &.va-button--with-right-icon {
      padding-right: $btn-with-icon-wrapper-padding-nrm;

      .va-button__content__title {
        padding-right: $btn-with-icon-content-padding-nrm;
      }
    }

    &.va-button--outline {
      line-height: $btn-line-height-nrm - 2 * $btn-border-outline;
    }
  }
}
</style>
