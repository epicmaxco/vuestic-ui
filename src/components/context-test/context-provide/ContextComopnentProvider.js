import { ContextProviderKey } from './ContextPlugin'

/**
 * Component for add local config through inject / provide
 * If this component contains top-level local config, inject takes a empty value.
 * In provide returns list with one elements: this config
 * In other cases component will take a list of local configs and return a new list with injected configs and property config
 *
 * In method provide will return a new object with ContextProviderKey and a new list configs
 */
export default {
  name: 'va-context-component-provider',
  inject: {
    _$configs: {
      from: [ContextProviderKey],
      default: () => [],
    },
  },
  props: {
    config: {
      type: Object,
      required: true,
    },
  },
  provide () {
    const observableConfig = this.config
    const newConfig = this._$configs ? [ ...this._$configs, observableConfig ] : []

    return { [ContextProviderKey]: newConfig }
  },
  render () {
    return this.$slots.default || null
  },
}
