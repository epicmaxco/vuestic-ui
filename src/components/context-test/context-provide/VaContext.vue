<script>
import { ContextProviderKey } from './ContextPlugin'

// This component just attaches local config to injected config chain,
// then passes it down via provide.
export default {
  name: 'VaContext',
  inject: {
    _$configs: {
      from: [ContextProviderKey],
      default: () => [],
    },
  },
  props: {
    config: {
      type: Object,
      required: true,
    },
  },
  provide () {
    const newConfig = this._$configs ? [...this._$configs, this.config] : []

    return { [ContextProviderKey]: newConfig }
  },
  render () {
    return this.$slots.default || null
  },
}
</script>
