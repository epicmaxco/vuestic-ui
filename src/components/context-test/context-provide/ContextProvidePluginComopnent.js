import Vue from 'vue'
import { ContextProviderPluginKey } from './ContextProvidePlugin'

export default {
  name: 'va-context-config-component',
  inject: {
    _$configs: {
      from: [ContextProviderPluginKey],
      default: () => [],
    },
  },
  props: {
    config: {
      type: Object,
      require: true,
    },
  },
  provide () {
    const newConfig = (this._$configs && this._$configs.length) ? [ ...this._$configs, Vue.observable(this.config) ] : []

    return { [ContextProviderPluginKey]: newConfig }
  },
  render () {
    return this.$slots.default || null
  },
}
