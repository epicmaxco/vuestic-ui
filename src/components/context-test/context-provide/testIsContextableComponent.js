import { mount } from '@vue/test-utils'
import VaContext from './VaContext'

// Check passed props are contextable.
// componentProps could be used to pass down required props.
export function testIsContextableComponent (componentOptions, contextProps, componentProps) {
  const wrapperComponentOptions = {
    render (createElement) {
      return createElement(
        VaContext,
        { props: { config: { [componentOptions.name]: contextProps } } },
        [
          createElement(componentOptions, { props: componentProps }),
        ],
      )
    },
  }

  const wrapper = mount(wrapperComponentOptions)
  const component = wrapper.vm.$children[0].$children[0]

  Object.keys(contextProps).forEach(prop => {
    if (component[`c_${prop}`] !== contextProps[prop]) {
      throw new Error(`Prop ${prop} doesn't appear to take value from context.`)
    }
  })
}
