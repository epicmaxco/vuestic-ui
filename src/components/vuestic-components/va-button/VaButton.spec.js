import Vue from 'vue'
import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import VaButton from './VaButton'

import { ColorThemePlugin } from '../../../services/ColorThemePlugin'
import { ContextPlugin } from '../../context-test/context-provide/ContextPlugin'

Vue.use(ColorThemePlugin)
Vue.use(ContextPlugin, {})

describe('VaButton', () => {
  /* 1. Default button */

  it('button without defined props', () => {
    const wrapper = shallowMount(VaButton)

    expect(wrapper.find(VaButton).classes()).toContain('va-button--default')
  })

  /* 2. Button sizes */

  it('normal button', () => {
    const wrapper = shallowMount(VaButton)

    expect(wrapper.find(VaButton).classes()).toContain('va-button--normal')
  })

  it('large button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { size: 'large' },
    })
    expect(wrapper.find(VaButton).classes()).toContain('va-button--large')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('small button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { size: 'small' },
    })
    expect(wrapper.find(VaButton).classes()).toContain('va-button--small')
    expect(wrapper.html()).toMatchSnapshot()
  })

  /* 3. Button types */

  it('default button', () => {
    const wrapper = shallowMount(VaButton)

    expect(wrapper.is('button')).toBe(true)
  })

  it('a button with defined href', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        href: 'http://epic-spinners.epicmax.co/',
      },
    })
    expect(wrapper.is('a')).toBe(true)
  })

  it('a button with defined target', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        target: '_blank',
      },
    })
    expect(wrapper.is('a')).toBe(true)
  })

  it('a button with defined href and target', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        href: 'http://epic-spinners.epicmax.co/',
        target: '_blank',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('router-link button with defined to property', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        to: { name: 'charts' },
      },
      stubs: {
        RouterLink: RouterLinkStub,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('router-link button with defined to and active class properties', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        to: { name: 'charts' },
        activeClass: 'va-button--active',
      },
      stubs: {
        RouterLink: RouterLinkStub,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  /* 4. Disabled buttons */

  it('disabled button', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: { disabled: true },
    })

    expect(wrapper.find(VaButton).classes()).toContain('va-button--disabled')
    expect(wrapper.html()).toMatchSnapshot()
  })

  /* 5. Buttons with icons */

  it('button with left icon', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        icon: 'clear',
      },
    })
    expect(wrapper.find(VaButton).classes())
      .toContain('va-button--with-left-icon')
  })

  it('button with right icon', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        iconRight: 'create',
      },
    })
    expect(wrapper.find(VaButton).classes())
      .toContain('va-button--with-right-icon')
  })

  it('button with both icons', () => {
    const wrapper = shallowMount(VaButton, {
      propsData: {
        icon: 'clear',
        iconRight: 'create',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
