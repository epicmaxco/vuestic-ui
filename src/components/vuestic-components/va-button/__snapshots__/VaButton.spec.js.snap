// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaButton a button with defined href and target 1`] = `
<a type="button" href="http://epic-spinners.epicmax.co/" target="_blank" tabindex="0" class="va-button va-button--default va-button--without-title va-button--normal" style="color: rgb(255, 255, 255); box-shadow: 0 0.125rem 0.19rem 0 rgba(64, 229,131, 0.4);">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</a>
`;

exports[`VaButton button with both icons 1`] = `
<button type="button" href="" target="" tabindex="0" class="va-button va-button--default va-button--without-title va-button--with-left-icon va-button--with-right-icon va-button--normal" style="color: rgb(255, 255, 255); box-shadow: 0 0.125rem 0.19rem 0 rgba(64, 229,131, 0.4);">
  <div class="va-button__content">
    <va-icon-stub color="" size="medium" name="clear" fixedwidth="true" rotation="" tag="i" class="va-button__content__icon va-button__content__icon-left"></va-icon-stub>
    <!---->
    <va-icon-stub color="" size="medium" name="create" fixedwidth="true" rotation="" tag="i" class="va-button__content__icon va-button__content__icon-right"></va-icon-stub>
  </div>
</button>
`;

exports[`VaButton disabled button 1`] = `
<button disabled="disabled" type="button" href="" target="" tabindex="0" class="va-button va-button--disabled va-button--without-title va-button--normal" style="color: rgb(255, 255, 255); box-shadow: 0 0.125rem 0.19rem 0 rgba(64, 229,131, 0.4);">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton large button 1`] = `
<button type="button" href="" target="" tabindex="0" class="va-button va-button--default va-button--without-title va-button--large" style="color: rgb(255, 255, 255); box-shadow: 0 0.125rem 0.19rem 0 rgba(64, 229,131, 0.4);">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;

exports[`VaButton router-link button with defined to and active class properties 1`] = `
<a type="button" href="" target="" tabindex="0" class="va-button va-button--default va-button--without-title va-button--normal" style="color: rgb(255, 255, 255); box-shadow: 0 0.125rem 0.19rem 0 rgba(64, 229,131, 0.4);">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</a>
`;

exports[`VaButton router-link button with defined to property 1`] = `
<a type="button" href="" target="" tabindex="0" class="va-button va-button--default va-button--without-title va-button--normal" style="color: rgb(255, 255, 255); box-shadow: 0 0.125rem 0.19rem 0 rgba(64, 229,131, 0.4);">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</a>
`;

exports[`VaButton small button 1`] = `
<button type="button" href="" target="" tabindex="0" class="va-button va-button--default va-button--without-title va-button--small" style="color: rgb(255, 255, 255); box-shadow: 0 0.125rem 0.19rem 0 rgba(64, 229,131, 0.4);">
  <div class="va-button__content">
    <!---->
    <!---->
    <!---->
  </div>
</button>
`;
